std/json.d

```diff
@@ -359,6 +359,90 @@ struct JSONValue
         return type == JSONType.null_;
     }
 
+    /***
+     * Generic type value getter
+     * A convenience getter that returns this `JSONValue` as the specified D type.
+     * Note: only arithmetic, `bool`, `string`, `JSONValue` and `JSONValue[]` types are accepted
+     * Throws: `JSONException` if `T` cannot hold the contents of this `JSONValue`
+     */
+    @property inout(T) get(T)() inout const pure @safe
+    {
+        static if (is(T : string))
+        {
+            return str;
+        }
+        else static if (is(T : bool))
+        {
+            return boolean;
+        }
+        else static if (isFloatingPoint!T)
```

Comment:
Off topic for this PR: I wonder what would be the best way to support user-defined types, such as bigint, decimal or checkedint, without whitelisting them one by one here.

Reply:
If you ask me, it seems possible to make something like `return T(floating);` but I'm not entirely sure if it's a good solution both in style and performance. It would properly initialize the T with value if that is possible though.